src/engine/IndexScan.cpp

```diff
@@ -291,9 +317,18 @@ void IndexScan::determineMultiplicities() {
     }
   }();
   for ([[maybe_unused]] size_t i :
-       ql::views::iota(multiplicity_.size(), getResultWidth())) {
+       ql::views::iota(multiplicity_.size(),
+                       std::max(multiplicity_.size(), getResultWidth()))) {
```

Comment:
Under what kind of circumstances is the result width smaller than the `multiplicity_` vector? Shouldn't it be always the same size? Or perhaps <= getResultWidth()?